.github/workflows/integration-test.yml

```diff
@@ -12,21 +12,49 @@ env:
 jobs:
   tests:
     name: Run integration tests
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest , windows-latest ]
+        jdk: [ 11, 17 ]
+    runs-on: ${{ matrix.os }}
+
     steps:
       - name: Checkout Branch
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.jdk }}
 
       - uses: browser-actions/setup-geckodriver@latest
       - run: geckodriver --version
 
-      - uses: browser-actions/setup-firefox@latest
-      - run: firefox --version
-          
-      - name: Download OpenSearch Security Plugin
+      - name: Set up Firefox browser for Linux
+        if: ${{ runner.os == 'Linux' }}
+        uses: browser-actions/setup-firefox@latest
+        with:
+          browser: firefox
+          version: latest
+
+      # Browser-action version does not work on Windows
+      - name: Set up Firefox browser for Windows
+        if: ${{ runner.os == 'Windows' }}
+        uses: RyanL1997/setup-browser@main
+        with:
+          browser: firefox
+          version: latest
+
+      - name: Download OpenSearch Security Plugin on Linux
```

Comment:
Lets reduce the duplication between the cypress and integration tests workflow.  There are a couple ways this can be done.

- Refactor the common code into a new action like install-dashboards, then trim down these workflow files
- Merge both workflow files into a single file with a matrix that selects the type of integration test, and include if statements for the cypress vs jest runners

I would prefer the first option to make a common action they reuse, that could be re-used by other teams, what do you think?

Reply:
Thanks for the review! I have created a new action for the plugin download and script setup~